### ISSUE TITLE
Hoodoo::TransientStore Regression in Mock response for Test environments.

### ISSUE THREAD

**Issue by whithajess** [LoyaltyNZ/hoodoo#212]
Moving from `1.14.0` -> `1.15.0` seems to break some previous functionality from **Hoodoo::Services::Session** as far as I can tell this code:
```
if Hoodoo::Services::Middleware.environment.test? && MockDalliClient.bypass? == false
  return MockDalliClient.new
end
```
was removed see `git diff v1.14.0..v1.15.0 lib/hoodoo/services/services/session.rb` the impact being that previously when in a test environment you could give nothing to the **Hoodoo:Services:Session** initialiser for `memcached_host` allowing you to run tests without Memcached installed.

The replacing functionality returns exceptions raised from the new method in the same class 
```
def get_store
...
raise "Hoodoo::Services::Session\#get_store: Cannot connect to Memcached at '#{ host }': #{ exception.to_s }"
```
This is technically a breaking change but only really affects tests, and I am not sure if the functionality needs to be put back but we should at least make it known in the `CHANGELOG`?

**Comment by pond** [comment]
Tackled by https://github.com/LoyaltyNZ/hoodoo/pull/213.

**Comment by pond** [comment]
Fixed by #213, merged and released as version 1.16.1.
